src/vm/i386/jithelp.S

```diff
@@ -0,0 +1,782 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+.intel_syntax noprefix
+#include "unixasmmacros.inc"
+#include "asmconstants.h"
+
+.macro JIT_FetchHelper Reg
+LEAF_ENTRY JIT_Fetch\Reg, _TEXT
+      mov \Reg, [esp]
+      ret
+LEAF_END JIT_Thunk, _TEXT
+.endm
+
+JIT_FetchHelper EAX
+JIT_FetchHelper EBX
+JIT_FetchHelper EDX
+
+.macro PREPARE_EXTERNAL_VAR_EAX Name
```

Comment:
This should be in the unixasmmacros file for x86. You will need it in another helper, the ResolveWorkerChainLookupAsmStub too. And for other architectures, it is in there too.
I would also like to name the macro JIT_Fetch and the JIT_FetchHelper functions to something more informative, like GetCurrentEip. Or you can get rid of these macros completely using a trick that clang or intel C++ compiler do:
```asm
.att_syntax
    call  0f
0:
    pop %ebx
1:
    addl  $_GLOBAL_OFFSET_TABLE_+(b1 - b0), %ebx
    movl  C_FUNC(\Name)@GOT(%ebx), %ebx
    movl  (%ebx), %ebx
.endm